Makefile

```diff
@@ -4,7 +4,13 @@ PACKAGES=$(shell go list ./...)
 BUILDDIR ?= $(CURDIR)/build
 
 BUILD_TAGS?=tendermint
-VERSION := $(shell git describe --always)
+
+ifneq ($(shell git branch --show-current),)
```

Comment:
Shouldn't this compare with `master` or something?

Reply:
Ah, I guess we're checking that we're not at a branch tip. This is going to break if we tag the tip of a branch.

Reply:
If we tag a branch then someone uses the tag it won't show the branch name under `git branch --show-current`. I doubled checked on v0.34.0-rc2 which was taken off master and it worked as intended.

Reply:
```
$ git checkout -b foo
Switched to a new branch 'foo'
$ git tag vFOO
$ git branch --show-current
foo   # Not vFOO!
```

Reply:
Of course, as you say, checking out `vFOO` explicitly works even when at the branch tip. Maybe that's ok, but should be documented with a comment perhaps.

Reply:
You are still on branch foo. I usually checkout tags so when I tagged and checked it out it worked.

Reply:
Do you have suggestions on how to better it? I don't off the top of my head but willing to modify/change it.

Reply:
No, I think that's fine (alternative would be to use some sort of branch name pattern matching, but that's brittle too). Let's just make sure people and tooling is aware so we don't mess it up.

Reply:
Why not just a good ole' `git describe --always | sed 's/^v//'`?

Reply:
What if my branch is "very-nice"? 😀